components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/ModelValidatorsTest.java

```diff
@@ -0,0 +1,131 @@
+            "hibernate.validator.fail_fast", "true").buildValidatorFactory().getValidator();
+
+
+    @Test(dataProvider = "dp-checkValidScopeFormat", dataProviderClass = ValidatorTestDataProvider.class)
+    public void checkValidScopeFormat(String claimsString) throws ParseException {
+
+        sampleRequestObject.setClaimSet(JWTClaimsSet.parse(claimsString));
+        Set<ConstraintViolation<SampleRequestObject>> violations = validator.validate(sampleRequestObject);
+        String violation = violations.stream().findFirst().map(ConstraintViolation::getMessage).orElse(null);
+        assertNull("Valid scope formats should pass", violation);
+    }
+
+    @Test(dataProvider = "dp-checkValidSingleScopes", dataProviderClass = ValidatorTestDataProvider.class)
+    public void checkValidSingleScopes(String claimsString) throws ParseException {
+
+        sampleRequestObject.setClaimSet(JWTClaimsSet.parse(claimsString));
+        Set<ConstraintViolation<SampleRequestObject>> violations = validator.validate(sampleRequestObject);
+        String violation = violations.stream().findFirst().map(ConstraintViolation::getMessage).orElse(null);
+        assertNull("Valid single scope should pass", violation);
+    }
+
+    @Test
+    public void checkMandatoryParamsValidationFailing() {
+
+        SampleDifferentClass sampleRequestObject = new SampleDifferentClass();
+        sampleRequestObject.setName("name");
+        sampleRequestObject.setMale(true);
+
+        Set<ConstraintViolation<SampleDifferentClass>> violations = validator.validate(sampleRequestObject);
+        String violation = violations.stream().findFirst().map(ConstraintViolation::getMessage).orElse(null);
+        assertEquals("age failed", violation);
+
+        //
+        sampleRequestObject = new SampleDifferentClass();
+        sampleRequestObject.setName("name");
+        sampleRequestObject.setAge(70);
+
+        violations = validator2.validate(sampleRequestObject);
+        violation = violations.stream().findFirst().map(ConstraintViolation::getMessage).orElse(null);
+        assertEquals("male failed", violation);
+
+        //
+        sampleRequestObject = new SampleDifferentClass();
+        sampleRequestObject.setMale(true);
+        sampleRequestObject.setAge(70);
+
+        violations = validator3.validate(sampleRequestObject);
+        violation = violations.stream().findFirst().map(ConstraintViolation::getMessage).orElse(null);
+        assertEquals("name failed", violation);
+
+    }
+
+    @Test
+    public void checkOpenBankingValidator() {
+
+        SampleDifferentClass sampleRequestObject = new SampleDifferentClass();
+        sampleRequestObject.setName("name");
+        sampleRequestObject.setMale(true);
+
+        String violation = BFSIValidator.getInstance().getFirstViolation(sampleRequestObject);
+        assertEquals("age failed", violation);
+
+        //
```

Comment:
```suggestion
```

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/13/commits/cd47ead8b0ab89e7914625db1aa8437e57dc4655